.github/workflows/gcc.yml

```diff
@@ -8,7 +8,7 @@ defaults:
     shell: bash -leo pipefail {0}
 
 env:
-  cache_key: gcc3  # The number (#) following the cache_key "gcc" is to flush Action cache.
+  cache_key: gcc4  # The number (#) following the cache_key "gcc" is to flush Action cache.
```

Comment:
Just for my learning purposes, why do we increase gcc from 3 to 4?   Similar question for intel in intel.yml.

Reply:
Good question.
There is a comment in this yaml file (reproduced below):
```
# A note on flushing Action cache and relevance to "cache_key" above.
# There is no way to flush the Action cache, and hence a number (#) is appended
# to the "cache_key" (gcc).
# If the dependencies change, increment this number and a new cache will be
# generated by the dependency build step "setup"
# There is a Github issue to force clear the cache.
# See discussion on:
# https://stackoverflow.com/questions/63521430/clear-cache-in-github-actions
```
Basically, if we change the version of the dependencies, we need to rebuild the cache for it to take effect.
The cache is then retained and reused in subsequent PR's if there is no update in the dependencies, saving time in the CI.  
We don't update the library versions often, so there is no need to touch this file (and increment the key) with every PR.
But when we do update, this cache needs to be rebuilt once.

Reply:
OK.  Let me digest this.   Thank you, @aerorahul